projects/fsastorefrontlib/src/cms-components/user-profile/guards/user-profile-guard.ts

```diff
@@ -26,17 +26,28 @@ export class UserProfileGuard implements CanActivate {
       filter(customer => !!customer),
       map(customer => {
         const oboCustomerId = route.params['customerId'];
-        if (
-          !!oboCustomerId &&
-          !customer.roles.includes(FSUserRole.SELLER) &&
-          customer.uid !== oboCustomerId
-        ) {
-          this.globalMessageService.add(
-            { key: 'organization.notification.noSufficientPermissions' },
-            GlobalMessageType.MSG_TYPE_WARNING
-          );
-          this.routingService.go({ cxRoute: 'home' });
+
+        if (oboCustomerId === '') {
+          this.routingService.go({
+            cxRoute: 'userProfile',
+            params: { customerId: customer.uid },
+          });
+        } else {
+          if (
```

Comment:
Maybe unite this if with previous one?
```
if (oboCustomerId === '' && customer.uid !== oboCustomerId && !customer.roles.includes(FSUserRole.SELLER))
```
Will this work as well?

Reply:
This approach won't work because the case where oboCustomerId is not empty and differs from the customer.uid will never be covered. 
It is important that we cover this case because the sellers need it to get the info from other customers that are assigned to them.